### PR TITLE
Add more ray_tracing coverage

### PR DESCRIPTION
**Pull request by loganharbour**
Refs #16028

### CODE REVIEW

**Comment by loganharbour** [comment]
@GiudGiud another short one

File: modules/ray_tracing/test/tests/userobjects/ray_tracing_study/errors/tests [hunk 1]
```diff
@@ -173,21 +181,64 @@
...
+      detail = 'when the study is set to execute with residual kernels and non-residual kernels,'
+    []
+    [eigenvalue]
+      type = RunException
+      input = 'errors.i'
+      cli_args = 'UserObjects/study/execute_on=PRE_KERNELS
+                  Executioner/type=Eigenvalue'
+      expect_err = 'is not supported for an eigenvalue solve.'
+      allow_test_objects = true
+      detail = 'when the study is set to execute with residual kernels and an eigenvalue executioner, and'
```
Comment (GiudGiud) [inline modules/ray_tracing/test/tests/userobjects/ray_tracing_study/errors/tests:225]:
```suggestion
      detail = 'when the study is set to execute with residual kernels and an eigenvalue executioner,'
```

File: modules/ray_tracing/src/base/RayTracingObject.C [hunk 2]
```diff
@@ -32,6 +32,9 @@ RayTracingObject::validParams()
   // Execute flags don't make sense here because the RayTracingStudy calls this object
   params.suppressParameter<ExecFlagEnum>("execute_on");
 
+  // Allow rays to change
+  params.declareControllable("rays");
```
Comment (GiudGiud) [inline modules/ray_tracing/src/base/RayTracingObject.C:36]: is there a test for it?

Comment (loganharbour) [reply modules/ray_tracing/src/base/RayTracingObject.C:36]: Nope - removing it for now

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**